src/vllm_tgis_adapter/grpc/grpc_server.py

```diff
@@ -512,8 +512,17 @@ def _convert_output(  # noqa: PLR0913
         return response
 
     @staticmethod
-    def request_id(context: ServicerContext) -> str:  # noqa:  ARG004
-        return uuid.uuid4().hex
+    def request_id(context: ServicerContext) -> str:
+        metadata = context.invocation_metadata()
+        if not metadata:
+            return uuid.uuid4().hex
+
+        correlation_id = dict(metadata).get("x-correlation-id")
```

Comment:
We have the same "magic string" in `log.py`:
https://github.com/opendatahub-io/vllm-tgis-adapter/blob/20db20b3ba4b184fde441b6ea5a9df6ee6894850/src/vllm_tgis_adapter/tgis_utils/logs.py#L137

Would it be worthwhile to extract it into a constant?

Reply:
Probably not